integrations/langfuse/tests/test_tracer.py

```diff
@@ -397,3 +404,61 @@ def test_init_with_tracing_disabled(self, monkeypatch, caplog):
 
             LangfuseTracer(tracer=MockTracer(), name="Haystack", public=False)
             assert "tracing is disabled" in caplog.text
+
+    def test_context_cleanup_after_nested_failures(self):
+        """
+        Test that tracer context is properly cleaned up even when nested operations fail.
+
+        This test addresses a critical bug where failing nested operations (like inner pipelines)
+        could corrupt the tracing context, leaving stale spans that affect subsequent operations.
+        The fix ensures proper cleanup through try/finally blocks.
+
+        Before the fix: context would retain spans after failures (length > 0)
+        After the fix: context is always cleaned up (length == 0)
+        """
+
+
+        @component
+        class FailingParser:
+            @component.output_types(result=str)
+            def run(self, data: str):
+                # This will fail with ValueError when data is not valid JSON
+                parsed = json.loads(data)
+                return {"result": parsed["key"]}
+
+        @component
+        class ComponentWithNestedPipeline:
+            def __init__(self):
+                # This simulates IntentClassifier's internal pipeline
+                self.internal_pipeline = Pipeline()
+                self.internal_pipeline.add_component("parser", FailingParser())
+
+            @component.output_types(result=str)
+            def run(self, input_data: str):
+                # Run nested pipeline - this is where corruption occurs
+                result = self.internal_pipeline.run({"parser": {"data": input_data}})
+                return {"result": result["parser"]["result"]}
+
+        tracer = LangfuseConnector("test")
+
+        main_pipeline = Pipeline()
+        main_pipeline.add_component("nested_component", ComponentWithNestedPipeline())
+        main_pipeline.add_component("tracer", tracer)
+
+        # Test 1: First run will fail and should clean up context
+        try:
+            main_pipeline.run({"nested_component": {"input_data": "invalid json"}})
+        except Exception:
+            pass  # Expected to fail
+
+        # Critical assertion: context should be empty after failed operation
+        assert len(tracer.tracer._context) == 0
+
+        # Test 2: Second run should work normally with clean context
+        try:
+            result = main_pipeline.run({"nested_component": {"input_data": '{"key": "valid"}'}})
+        except Exception:
+            assert False, "Second run should not fail"
```

Comment:
If this is expected to run we can get rid of the try except block. If the run fails the test will fail with an exception